plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StartTask.java

```diff
@@ -0,0 +1,354 @@
+        // version is not available as a maven artifact
+//        getProject().getConvention().getPlugin(JavaPluginConvention.class).getSourceSets()
+//                .getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput().getClassesDir();
+        final List<Path> sourcePaths = Arrays.asList(getProject().getBuildDir().toPath().resolve("classes/java/main"),
+                getProject().getBuildDir().toPath().resolve("resources/main"));
+        getLogger().info("Source paths: {}", sourcePaths);
+
+        final SwarmExecutor executor = new SwarmExecutor()
+                .withModules(getModules())
+                .withProperty(BootstrapProperties.APP_NAME, finalName)
+                .withClassPathEntries(sourcePaths)
+                .withClassPathEntries(getClassPathEntries(sourcePaths, true));
+
+        if (extension.getMainClassName() != null) {
+            getLogger().info("With main class " + extension.getMainClassName());
+            executor.withMainClass(extension.getMainClassName());
+        } else {
+            getLogger().info("With default main class.");
+            executor.withDefaultMainClass();
+        }
+
+        return executor;
+    }
+
+    private File getArchivePath() {
+        final Jar task = ThorntailUtils.getArchiveTask(getProject());
+        final File archivePath = task.getArchivePath();
+        getLogger().info("Archive path: {}", archivePath.getAbsolutePath());
+        return archivePath;
+    }
+
+    private File getUberJarArchivePath() {
+        final File outputFile = getProject().getTasks().withType(PackageTask.class)
+                .getByName(PackagePlugin.THORNTAIL_PACKAGE_TASK_NAME).getOutputFile();
+        getLogger().info("Uber-JAR archive path: {}", outputFile.getAbsolutePath());
+        return outputFile;
+    }
+
+    private List<Path> getModules() {
+        List<Path> modules = extension.getModules().stream().map(File::toPath).collect(Collectors.toList());
+        getLogger().info("With modules:\n  {}",
+                modules.stream().map(Path::toString).collect(Collectors.joining(INDENTATION_DELIMITER)));
+        return modules;
+    }
+
+    private List<Path> getClassPathEntries(List<Path> sourcePaths, boolean scanDependencies) {
+        Collection<ArtifactSpec> allDependencies = GradleToolingHelper
+                .toDeclaredDependencies(extension.getDependencies()).getRuntimeExplicitAndTransientDependencies();
+        getLogger().info("Original dependencies including transitive:\n  {}",
+                allDependencies.stream().map(ArtifactSpec::toString).collect(Collectors.joining(INDENTATION_DELIMITER)));
+
+        final List<Path> classpathEntries = allDependencies.stream()
+                .map(artifactSpec -> artifactSpec.file.toPath())
+                .collect(Collectors.toList());
+
+        final boolean hasThorntailDeps = allDependencies.stream()
```

Comment:
Do you want to rename this variable from `hasThorntailDeps` to `hasBootstrapDeps` or something similar? That way, when we refactor the plugin code to include only the essential pieces, it would be easier to read this code. What do you think?